dask/array/routines.py

```diff
@@ -1326,7 +1327,8 @@ def average(a, axis=None, weights=None, returned=False):
             # setup wgt to broadcast along axis
             wgt = broadcast_to(wgt, (a.ndim - 1) * (1,) + wgt.shape)
             wgt = wgt.swapaxes(-1, axis)
-
+        # if there is a masked array
+        wgt = wgt * (~getmaskarray(a))
```

Comment:
Rather than doing this here, I'd rather that a thin wrapper around the existing functionality was implemented in the `da.ma` module. numpy's average function doesn't work for masked arrays, but `np.ma.average` does.

The best way to handle this might be to extract this full method into a helper method that takes an additional `is_masked` boolean (or something), and optionally applies this line.

```suggestion
        if is_masked:
            from .ma import getmaskarray
            wgt = wgt * (~getmaskarray(a))
```

Then you'd define `da.average` with `is_masked` set to False, and `da.ma.average` with `is_masked` set to True.

Reply:
This would also fix the numpy version import issue, as this line would only execute if called from within `da.ma.average`.

Reply:
Updated with your suggestion.